docs/articles/baseplatform/reproducible-yocto-builds.rst

```diff
@@ -0,0 +1,134 @@
+Reproducible Yocto builds
+=========================
+
+Intro
+-----
+
+This document describes how to configure a Yocto build so that it can be reproduced,
+and how to reproduce such build.
+The reproduction requires that additional artifacts are archived.
+
+
+The AUTOREV problem
+-------------------
+
+A bitbake recipe specifies where the source code of a package should be downloaded from.
+Because there are several ways to do this, a recipe might not specify a specific version but rather
+the head of some branch or a latest release.
+
+Even though this is not recommended, there is no guarantee that developers will not write recipes
+to use ``AUTOREV`` to point to latest source revision.
+This of course affects reproducibility as the maintainer of some upstream project may update the
+head of the branch or the latest release between builds so that bitbaking the same version of
+a recipe yields different results.
+In addition, in cases when reproducibility of a Yocto build is required, the actual availability of
+the upstream source code packages might cause a problem, e.g. if a tar archive is moved to some
+other URL.
+
+The general solution to guarantee reproducibility is to let the Yocto build environment create tar
+archives of the downloaded sources and place these in a local mirror during a build.
+This local mirror can then be used when reproducing the build, a solution that works in most cases.
+The exception is when a package's source code is stored in a git repository and the recipe for the
```

Comment:
I still don't understand this: "It's not clear to me why this wouldn't work: if Yocto makes a tar archive of the git repo, than it cannot change, can it?"

Reply:
It's quite clear Mardy, Yocto will create the tartballs of the git repos and they won't change.
Yocto won't create a tarball if the recipe have URL point to git repo and SRCREV is AUTOREV.

because Yocto knows that Internet access always needed to retrieve the latest release from the remote git repo.
This behavior is clarified in line 40 and 41, and showing potential error messages you will get when that happens.

Reply:
Ah, now it's clear :-) But it's not written as clearly, not even in lines 40 and 41: I suggest you add this to the end of this paragraph (line 32): "...point to a specific commit but a tag: in that case, Yocto won't create a tarball."